File: sales-portal-robot-tests/data/schemas/products/get_all_products_schema.py
"""Get all products response JSON schema."""

from __future__ import annotations

from typing import Any

from data.schemas.core_schema import OBLIGATORY_FIELDS_SCHEMA, OBLIGATORY_REQUIRED_FIELDS
from data.schemas.products.product_schema import PRODUCT_SCHEMA

GET_ALL_PRODUCTS_SCHEMA: dict[str, Any] = {
    "type": "object",
    "properties": {
        "Products": {
            "type": "array",
            "items": PRODUCT_SCHEMA,
        },
        **OBLIGATORY_FIELDS_SCHEMA,
    },
    "required": ["Products", *OBLIGATORY_REQUIRED_FIELDS],
}

File: sales-portal-robot-tests/exmple_js_project/src/data/schemas/products/getAllProducts.schema.ts
import { obligatoryFieldsSchema, obligatoryRequiredFields } from "../core.schema";
import { productSchema } from "./product.schema";

export const getAllProductsSchema = {
  type: "object",
  properties: {
    Products: {
      type: "array",
      items: productSchema,
    },
    ...obligatoryFieldsSchema,
  },
  required: ["Products", ...obligatoryRequiredFields],
};
